rules/prefer-module.js

```diff
@@ -198,6 +209,94 @@ const isTopLevelReturnStatement = node => {
 	return true;
 };
 
+const isParentLiteral = node => {
+	if (node?.type !== 'Literal') {
+		return false;
+	}
+
+	return node.value === '.' || node.value === './';
+};
+
+const isImportMeta = node =>
+	node.type === 'MetaProperty'
+	&& node.meta.name === 'import'
+	&& node.property.name === 'meta';
+
+/** @returns {node is import('estree').NewExpression} */
+const isNewURL = node =>
+	node.type === 'NewExpression'
+	&& node.callee.type === 'Identifier'
+	&& node.callee.name === 'URL';
+
+/** @returns {node is import('estree').MemberExpression} */
+const isAccessPathname = node =>
+	node.type === 'MemberExpression'
+	&& getPropertyName(node) === 'pathname';
+
+function isCallNodeBuiltinModule(node, propertyName, nodeModuleName, sourceCode) {
+	if (node.type !== 'CallExpression') {
+		return false;
+	}
+
+	/** @type {{callee: import('estree').Expression}} */
+	const {callee} = node;
+	if (callee.type === 'MemberExpression') {
+		// Check for nodeModuleName.propertyName(...);
+		if (callee.object.type !== 'Identifier') {
+			return false;
+		}
+
+		if (getPropertyName(callee) !== propertyName) {
+			return false;
+		}
+
+		const specifier = getImportSpecifier(callee.object);
+		return specifier?.type === 'ImportDefaultSpecifier' || specifier?.type === 'ImportNamespaceSpecifier';
+	}
+
+	if (callee.type === 'Identifier') {
+		// Check for propertyName(...);
+		const specifier = getImportSpecifier(callee);
+
+		return specifier?.type === 'ImportSpecifier' && specifier.imported.name === propertyName;
+	}
+
+	return false;
+
+	function getImportSpecifier(node) {
```

Comment:
I understand this is a safe play, and you already added, it's good.

But there is another way to get these builtin modules now `process.getBuiltinModule`, maybe we can just simply check the callee is

```
fileURLToPath
url.fileURLToPath
``` 

?

Unless you want add support for `const url = process.getBuiltinModule('node:url')` and `const {fileURLToPath} = process.getBuiltinModule('node:url')`

Reply:
I think `fileURLToPath()` is mostly fine, but I think it's a bit dangerous to check `path.dirname()` only for identifiers.
`dirname()` seems like a common word, so I think it might be a user-created function.

I'll try changing it to check `process.getBuiltinModule()`.

---
By the way, this is the first time I've heard of the existence of `process.getBuiltinModule()` 😅.
Thanks for letting me know ❤️

Reply:
https://github.com/sindresorhus/eslint-plugin-unicorn/pull/2611 😄